extension/src/pipeline/quickPick.ts

```diff
@@ -18,7 +22,7 @@ export type PlotConfigData = {
   x: { file: string; key: string }
   template: string
   title: string
-  y: { file: string; key: string }
+  y: { [file: string]: string[] | string }
```

Comment:
[Q] Are we going to allow selecting multiple x values?

Reply:
> [Q] Are we going to allow selecting multiple x values?

Apologies, forgot to update #4654. Will do that now but in summary, quick picks do not retain the order that the user selects the items. Since we can't detect the order the user wants to add the values, it looks like were gonna just want to allow `x` to be a single value.

Reply:
👍🏻. NW we can wait to see if someone asks for it. Did you confirm this is not possible using both the `window.createQuickPick` and `window.showQuickPick` APIs?

Reply:
> Did you confirm this is not possible using both the window.createQuickPick and window.showQuickPick APIs?

Yes, as far as I can tell. None of the options show a way to change the order of the selected values. I also looked into some functions that get called on select but they didn't give us enough information for us to detect the order.

Reply:
How about something like this using `createQuickPick`?:

```
  new Promise(resolve => {
    const quickPick = createQuickPick(items, selectedItems, {
      ...DEFAULT_OPTIONS,
      canSelectMany: true,
      ...options
    })

    let orderedSelection: T[] = []
    quickPick.onDidChangeSelection(selectedItems => {
      const itemValues: T[] = []
      for (const item of selectedItems) {
        if (!orderedSelection.includes(item.value)) {
          orderedSelection.push(item.value)
        }
        itemValues.push(item.value)
      }
      orderedSelection = orderedSelection.filter(item =>
        itemValues.includes(item)
      )
    })
```